scripts/pull-postsubmit-retry.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The postsubmit job which publishes artifacts takes some time to complete.
+# This script retries pulling the postsubmit artifacts for a reasonable amount
+# of time and then errors if not found.
+
+set -euo pipefail
+
+# Wait up to 20 minutes for postsubmit artifacts
```

Comment:
Blocking a CI job for 20m waiting for another CI to finish seems... inefficient.
They can't be merged together?

If we do decide to do it this way, I would recommend adding some kind of progress reporting, like an echo for each attempt, so we can see in the log how many times it retried. And/or maybe a time print out to see how long we waited.

Reply:
This should actually be faster than merging the jobs together, as the postsubmit starts immediately after the commit is submitted. The periodic job could start at the same time as the postsubmit, but most likely would start some time after. So the build of the artifacts (postsubmit) should always have a head start on when the periodic job runs. I'm planning to investigate a similar approach for the other periodic jobs.

Sure, I can improve the output.

Reply:
Updated the output